XenAdmin/ConsoleView/VNCTabView.cs

```diff
@@ -1133,7 +1153,25 @@ private void toggleConsoleButton_Click(object sender, EventArgs e)
 
                     vncScreen.UseVNC = !vncScreen.UseVNC;
                     vncScreen.UserWantsToSwitchProtocol = true;
-                    
+
+                    if (Helpers.CreedenceOrGreater(targetHost))
+                    {
+                        if (!RDPEnabled)
+                        {
+                            ThreeButtonDialog d = new ThreeButtonDialog(
+                                new ThreeButtonDialog.Details(System.Drawing.SystemIcons.Warning, Messages.FORCE_ENABLE_RDP),
+                                   "EnableRDPonVM",
+                                new ThreeButtonDialog.TBDButton(Messages.YES, DialogResult.Yes),
+                                new ThreeButtonDialog.TBDButton(Messages.NO, DialogResult.No));
+                            if (d.ShowDialog(Program.MainWindow) == DialogResult.Yes)
+                            {
+                                Session session = source.Connection.DuplicateSession();
+                                Dictionary<string, string> _arguments = new Dictionary<string, string>();
+                                XenAPI.VM.call_plugin(session, source.opaque_ref, "guest-agent-operation", "request-rdp-on", _arguments);

```

Comment:
The design asked for us to switch RDP on, wait for it to become enabled, and then connect via RDP. I don't see that happening.